### DISCUSSION TITLE
Order by Multiple Fields

### DISCUSSION THREAD

**Discussion by fenos** [prisma/prisma#2884] [Q&A] [answered]
Hello, I hope you are all well and healthy.

I'm trying to achieve the easiest query ever with Prisma2 without success:

```ts
this.prisma.objective.findMany({
  orderBy: {
    createdByMemberId: 'asc',
    createdAt: 'desc',
  }
});
```

**Getting an error:**

```
ObjectiveOrderByInput needs exactly one argument, but you provided createdByMemberId and createdAt. Please choose one.
```

After seeing this error being returned when providing 2 fields or more I've entered in the "panic / anxious" zone at the moment, please help me out with this. 😄 

Any application in this planet (other than listing cute kittens in a page) will need to be able to sort by 2 or more columns

The current `PrismaClient Typescript` definition clearly allows providing multiple order by fields, so I'm suspecting I need to something in order to achieve this.

Can you please suggest how to achieve this?

I've built 50% of my application (mostly mutations) and now that I'm about to build the queries I've discovered that order by multiple columns is not supported 😭

**Comment by ryands17** [comment, marked as answer]
Hey @fenos :wave: 
Unfortunately this is currently only possible via [`prisma.queryRaw`](https://www.prisma.io/docs/reference/tools-and-interfaces/prisma-client/raw-database-access#queryraw). There is still an open request for this [here](https://github.com/prisma/prisma-client-js/issues/702) that you can follow.

**Comment by ludralph** [comment]
Hi @fenos,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
